src/main/scala/cromwell/engine/db/slick/ExecutionComponent.scala

```diff
@@ -68,10 +74,23 @@ trait ExecutionComponent {
     } yield execution.status)
 
   // see workflowExecutionsByStatuses
-  def executionStatusesByWorkflowExecutionIdAndCallFqns(workflowExecutionId: Int, callFqns: Traversable[String]) = {
-    for {
+  def executionStatusesByWorkflowExecutionIdAndCallFqns(workflowExecutionId: Int, scopeKeys: Traversable[ExecutionDatabaseKey]) = {
+    val scopeFqns = scopeKeys.map(_.fqn).toIterable
+    val scopeIndexes = scopeKeys.map( k => ExecutionIndex.fromIndex(k.index)).toIterable
+    val scopeID = scopeFqns zip scopeIndexes
+    /*
+     * FIXME: This is bad, there is probably a better way
+     * We want entries that have the right workflowID AND a (fqn, index) pair which is in scopeKeys
+     * Use workaround because slick does not supporting tuples with inSet ATM: https://github.com/slick/slick/pull/995 
+     */
+    val falseRep: Rep[Boolean] = false
+    (for {
       execution <- executions
-      if execution.workflowExecutionId === workflowExecutionId && (execution.callFqn inSet callFqns)
-    } yield execution.status
+      if execution.workflowExecutionId === workflowExecutionId
+    } yield execution) filter { exec =>
+      scopeID.map {
+        case (name, index) => exec.callFqn === name && exec.index === index
+      }.fold(falseRep)(_ || _)
```

Comment:
Hmm maybe, it's already pretty much the case though.
Except for the map fold that you want to replace with an exists if I understand. 
I tried that but there is this lifted embedding thing in slick so all types are actually Rep[T] in the queries, and I found it even more farfetched than this because I had to convert from Rep[T] to T and vice versa.
But I'm not an expert at all, probably @kshakir has better insight on this.